modules/EnsEMBL/Web/SpeciesDefs.pm

```diff
@@ -892,124 +907,139 @@ sub _parse {
           if (-e $image_path) {
-            $tree->{$url}{'SPECIES_IMAGE'} = $strain_image;
+            $tree->{$url}{'SPECIES_IMAGE'} = $parent_image;
             $no_image = 0;
           }
-          else {
-            ## Use the parent image for this strain
-            $image_path = $image_dir.'/'.$parent_image.'.png';
-            if (-e $image_path) {
-              $tree->{$url}{'SPECIES_IMAGE'} = $parent_image;
-              $no_image = 0;
-            }
-          }
         }
       }
-      if ($no_image) {
-        my $clade = $tree->{$url}{'SPECIES_GROUP'};
-        $tree->{$url}{'SPECIES_IMAGE'} = $labels->{$clade};
-      }
+    }
+    if ($no_image) {
+      my $clade = $tree->{$url}{'SPECIES_GROUP'};
+      $tree->{$url}{'SPECIES_IMAGE'} = $labels->{$clade};
+    }
 
-      ## Species-specific munging
-      if ($url ne "MULTI" && $url ne "databases") {
-                                       
-        my $display_name = $tree->{$url}{'SPECIES_DISPLAY_NAME'};
+    ## Species-specific munging
+    if ($url ne "MULTI" && $url ne "databases") {
+                                     
+      my $display_name = $tree->{$url}{'SPECIES_DISPLAY_NAME'};
              
-        # Populate taxonomy division using e_divisions.json template
-        push @{$species_to_assembly->{$display_name}}, $tree->{$url}->{'ASSEMBLY_VERSION'};
-        my $taxonomy = $tree->{$url}{'TAXONOMY'};
-        my $children = [];
-        my $other_species_children = [];
-        my @other_species = grep { $_->{key} =~ m/other_species/ } @{$tree->{'ENSEMBL_TAXONOMY_DIVISION'}->{child_nodes}};
-        $other_species[0]->{child_nodes} = [] if ($other_species[0] && !$other_species[0]->{child_nodes});
-
-        my $strain_name = $tree->{$url}{'SPECIES_STRAIN'};
-        my $strain_group = $tree->{$url}{'STRAIN_GROUP'};
-        my $group_name   = $tree->{$url}{'SPECIES_COMMON_NAME'};
-        my $species_key = $tree->{$url}{'SPECIES_URL'}; ## Key on actual URL, not production name
-
-        foreach my $node (@{$tree->{'ENSEMBL_TAXONOMY_DIVISION'}->{child_nodes}}) {
-          my $child = {
-            key             => $species_key,
-            scientific_name => $tree->{$url}{'SPECIES_SCIENTIFIC_NAME'},
-            common_name     => $tree->{$url}{'SPECIES_COMMON_NAME'},
-            display_name    => $tree->{$url}{'GROUP_DISPLAY_NAME'},
-            image           => $tree->{$url}{'SPECIES_IMAGE'},
-            is_leaf         => 'true'
-          };
-
-          if ($strain_group && $strain_name !~ /reference/) {
-            $child->{type} = $group_name . ' ' . $tree->{$url}{'STRAIN_TYPE'}. 's';
-          }
-          elsif($strain_group && $strain_name =~ /reference/) {
-            # Create display name for Reference species
-            my $ref_name = $tree->{$url}{'SPECIES_DISPLAY_NAME'} . ' '. $strain_name;
-            $child->{display_name} = $ref_name;
-          }
+      # Populate taxonomy division using e_divisions.json template
```

Comment:
I am not sure what has changed in this part of code since it looks the same just github seems to have shifted everything. Not sure if its the diff thats confused. @ens-ap5 can you confirm if there has been anything changed. It might be that you remove an if statement causing the identation to shift.

Reply:
I think it's the diff that's confusing - I had to break up the loop into two sections in order to add the merging of the generic collection values into the individual species trees at the correct point. Collection databases presumably have additional meta keys, equivalent to DEFAULTS.ini, that apply to all species in the collection.